store-rx2/src/main/kotlin/com/dropbox/store/rx2/RxStore.kt

```diff
@@ -0,0 +1,135 @@
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.reactive.asFlow
+import kotlinx.coroutines.rx2.asCoroutineDispatcher
+import kotlinx.coroutines.rx2.asFlowable
+import kotlinx.coroutines.rx2.await
+import kotlinx.coroutines.rx2.rxSingle
+
+/**
+ * Return a flow for the given key
+ * @param request - see [StoreRequest] for configurations
+ */
+fun <Key : Any, Output : Any> Store<Key, Output>.observe(request: StoreRequest<Key>): Flowable<StoreResponse<Output>> =
+    stream(request).asFlowable()
+
+/**
+ * Purge a particular entry from memory and disk cache.
+ * Persistent storage will only be cleared if a delete function was passed to
+ * [StoreBuilder.persister] or [StoreBuilder.nonFlowingPersister] when creating the [Store].
+ */
+fun <Key : Any, Output : Any> Store<Key, Output>.observeClear(key: Key): Single<Unit> =
+    rxSingle { clear(key) }
+
+/**
+ * Purge all entries from memory and disk cache.
+ * Persistent storage will only be cleared if a deleteAll function was passed to
+ * [StoreBuilder.persister] or [StoreBuilder.nonFlowingPersister] when creating the [Store].
+ */
+@ExperimentalStoreApi
+fun <Key : Any, Output : Any> Store<Key, Output>.observeClearAll(): Single<Unit> =
+    rxSingle { clearAll() }
+
+/**
+ * Creates a new [StoreBuilder] from a [Flowable] fetcher.
+ *
+ * Use when creating a [Store] that fetches objects in an websocket-like multiple responses
+ * per request protocol.
+ *
+ * @param fetcher a function for fetching a flow of network records.
+ */
+fun <Key : Any, Output : Any> rxFlowableStore(
+    fetcher: (key: Key) -> Flowable<Output>
+): StoreBuilder<Key, Output> = from { key: Key ->
+    fetcher.invoke(key).asFlow()
+}
+
+/**
+ * Creates a new [StoreBuilder] from a [Single] fetcher.
+ *
+ * Use when creating a [Store] that fetches objects from a [Single] source that emits one response
+ *
+ * @param fetcher a function for fetching a [Single] network response for a [Key]
+ */
+fun <Key : Any, Output : Any> rxSingleStore(
+    fetcher: (key: Key) -> Single<Output>
+): StoreBuilder<Key, Output> =
+    from { key: Key ->
+        fetcher.invoke(key).toFlowable().asFlow()
+
+    }
+
+/**
+ * Define what scheduler fetcher requests will be called on,
+ * if a scheduler is not set Store will use [GlobalScope]
+ */
+fun <Key : Any, Output : Any> StoreBuilder<Key, Output>.withScheduler(
+    scheduler: Scheduler
+): StoreBuilder<Key, Output> {
+    return scope(CoroutineScope(scheduler.asCoroutineDispatcher()))
+}
+
+/**
+ * Connects a (Non Flow) [Single] source of truth that is accessible via [reader], [writer],
+ * [delete], and [deleteAll].
+ *
+ * @see persister
+ */
+fun <Key : Any, Output : Any, NewOutput : Any> StoreBuilder<Key, Output>.withSinglePersister(
+    reader: (Key) -> Maybe<NewOutput>,
+    writer: (Key, Output) -> Single<Unit>,
+    delete: ((Key) -> Completable)? = null,
+    deleteAll: (() -> Completable)? = null
+): BuilderWithSourceOfTruth<Key, Output, NewOutput> {
```

Comment:
you should be able to just return `StoreBuilder` here and avoid the need to expose the concept of `sourceOfTruth` outside store. This (plus changing it for withFlowablePersister) should allow you to revert all the visibility changes

Reply:
done